src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCF.java

```diff
@@ -527,71 +903,165 @@ protected VariantContext cleanUpHighQualityVariant(final VariantContext result,
-            }
-            attrMap.put(GATKVCFConstants.VARIANT_DEPTH_KEY, varDP);
-            if (annotationEngine.getInfoAnnotations().stream()
-                    .anyMatch(infoFieldAnnotation -> infoFieldAnnotation.getClass().getSimpleName().equals("AS_QualByDepth"))) {
-                     final List<String> quals = new ArrayList<>();
-                for (final Allele alt : newAlleleSet) {
-                    if (alt.isReference()) {
-                        //GDB expects an empty field for ref
-                        continue;
-                    }
-                    if (alt.equals(Allele.NON_REF_ALLELE)) {
-                        quals.add("0");
-                        continue;
-                    }
-                    final GenotypesContext gc = AlleleSubsettingUtils.subsetAlleles(result.getGenotypes(),
-                            HomoSapiensConstants.DEFAULT_PLOIDY, result.getAlleles(), Arrays.asList(result.getReference(), alt), null,
-                            GenotypeAssignmentMethod.USE_PLS_TO_ASSIGN, result.getAttributeAsInt(VCFConstants.DEPTH_KEY,0));
-                    if (gc.get(0).hasPL()) {
-                        quals.add(Integer.toString(gc.get(0).getPL()[0]));
-                    } else {  //AlleleSubsettingUtils can no-call genotypes with super duper low GQs
-                        quals.add("0");
+    }
+
+    /**
+     * Add the newest raw mapping quality annotations to the annotation map
+     * @param originalVC    VariantContext that may contain deprecated mapping quality annotations
+     * @param newVCAttrMap   modified to add mapping quality raw annotations
+     */
+    private void updateMQAnnotations(final VariantContext originalVC, final Map<String, Object> newVCAttrMap) {
+        //all VCs should get new RAW_MAPPING_QUALITY_WITH_DEPTH_KEY, but preserve deprecated keys if present
+        if (!originalVC.hasAttribute(GATKVCFConstants.RAW_MAPPING_QUALITY_WITH_DEPTH_KEY)) {
+            //we're going to approximate depth for MQ calculation with the site-level DP (should be informative and uninformative reads),
+            //which is pretty safe because it will only differ if reads are missing MQ
+            final Integer rawMqValue = originalVC.hasAttribute(GATKVCFConstants.RAW_RMS_MAPPING_QUALITY_DEPRECATED) ?
+                    (int)Math.round(originalVC.getAttributeAsDouble(GATKVCFConstants.RAW_RMS_MAPPING_QUALITY_DEPRECATED, 0.0)) :
+                    (int)Math.round(originalVC.getAttributeAsDouble(VCFConstants.RMS_MAPPING_QUALITY_KEY, 60.0) *
+                            originalVC.getAttributeAsDouble(VCFConstants.RMS_MAPPING_QUALITY_KEY, 60.0) *
+                            originalVC.getAttributeAsInt(VCFConstants.DEPTH_KEY, 0));
+            newVCAttrMap.put(GATKVCFConstants.RAW_MAPPING_QUALITY_WITH_DEPTH_KEY,
+                    String.format("%d,%d", rawMqValue, originalVC.getAttributeAsInt(VCFConstants.DEPTH_KEY, 0)));
```

Comment:
Given that this gets called for every ref block would recommend not using the string.format() call here since the format caller is pathologically slow. I would recomment string.apppend() instead (which you can also get by simply doing "string" + "string")